### PR TITLE
The Preferencing/TGUIening

### PR DESCRIPTION
**Pull request by TheRealScarHomie**
## About The Pull Request

Makes the Pref menu like how it was AND improved! TGUI style and all that good stuff, also fixes the images on the Factions menu etc, ++ more stuff, TGUI is soul draining work so find out and see 👯.

image

oooh image oooh, you like image

### CODE REVIEW

File: mojave/code/modules/preferences/character.dm [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+/datum/preference/choiced/socks/init_possible_values()
+	return generate_values_for_underwear(GLOB.socks_list, list("human_r_leg", "human_l_leg"))
+
+/datum/preference/choiced/socks/apply_to_human(mob/living/carbon/human/target, value)
+	target.socks = value
+
+//Backpack/Storage
+
+#define TPACK "Travelpack"
+#define MBACKPACK "Millitary Backpack"
```
Comment (Hekzder) [inline mojave/code/modules/preferences/character.dm]:
```suggestion
#define MBACKPACK "Military Backpack"
```

File: mojave/items/clothing/other/ms13backpacks.dm [hunk 2]
```diff
@@ -33,13 +35,19 @@
...
 
 /obj/item/storage/ms13/ncr
 	name = "\improper NCR backpack"
 	desc = "A sturdy pack used by members of the NCR. It looks like it can hold a fair bit while still keeping a low profile."
 	icon_state = "ncr_pack"
-	inhand_icon_state = "ncr_pack"
-	worn_icon_state = "ncr_pack"
+
+/obj/item/storage/ms13/millitary
+	name = "millitary backpack"
```
Comment (Hekzder) [inline mojave/items/clothing/other/ms13backpacks.dm]:
```suggestion
	name = "military backpack"
```

File: mojave/items/clothing/other/ms13backpacks.dm [hunk 3]
```diff
@@ -33,13 +35,19 @@
...
 	slot_flags = ITEM_SLOT_BELT
 
 /obj/item/storage/ms13/ncr
 	name = "\improper NCR backpack"
 	desc = "A sturdy pack used by members of the NCR. It looks like it can hold a fair bit while still keeping a low profile."
 	icon_state = "ncr_pack"
-	inhand_icon_state = "ncr_pack"
-	worn_icon_state = "ncr_pack"
+
+/obj/item/storage/ms13/millitary
```
Comment (Hekzder) [inline mojave/items/clothing/other/ms13backpacks.dm]:
```suggestion
/obj/item/storage/ms13/military
```

File: mojave/items/clothing/other/ms13backpacks.dm [hunk 4]
```diff
@@ -33,13 +35,19 @@
...
+	name = "millitary backpack"
+	desc = "A tactical green storage device, made of sturdy materials and has enough pockets to hold anything you would ever need for wasteland combat."
+	icon_state = "millitary"
```
Comment (Hekzder) [inline mojave/items/clothing/other/ms13backpacks.dm]:
```suggestion
	icon_state = "military"
```

File: mojave/items/clothing/other/ms13backpacks.dm [hunk 5]
```diff
@@ -33,13 +35,19 @@
...
+
+/obj/item/storage/ms13/travel
+	name = "travelpack"
+	desc = "The roamers comapnion, a tall handcrafted pack, for storing all the essentials of wasteland livin'."
```
Comment (Hekzder) [inline mojave/items/clothing/other/ms13backpacks.dm]:
```suggestion
	desc = "The roamer's companion. A tall handcrafted pack for storing all the essentials of wasteland livin'."
```

File: mojave/code/modules/preferences/character.dm [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+#define SATCHEL "Satchel"
+#define SACK "Sack"
```
Comment (Hekzder) [inline mojave/code/modules/preferences/character.dm]: The storage sack, and generally having a belt slot storage item on spawn is not really something we're wanting to do right now. The item itself is fine to stay in but shouldn't be given to anyone on spawn for a variety of reasons, chiefly that it interferes with starting equipment for **a lot** of roles. But also that it has some _storage balance concerns_ that go against **THE VISION**

**Review by Hekzder — CHANGES_REQUESTED** [review]
This review is mostly just related to backpack stuff that I saw. Most of it is grammar or spelling related. Military is spelled with one L. 

The sack belt item thing is something I talked about with Original when it came to the gameplay concerns, but also like I mentioned in the comment about the sack, having a belt slot storage item as a choice on spawn interferes with a lot of roles starting equipment. In general it's not something we should really mess with right now, but we can maybe implement the sack as an item people can find in the world soon.

**Comment by TheRealScarHomie** [comment]
ok all done

**Comment by TheRealScarHomie** [comment]
ok all done

**Review by Hekzder — CHANGES_REQUESTED** [review]
Travelpack is missing an inventory state, currently the inventory.dmi icon it has is identical to the world state. 

When choosing the military backpack in character setup, the icon shows a sack instead. 

The NCR do not get their faction specific backpacks and instead get the chosen backpack in character setup. Faction backpack choice may remedy this if you can get it set up? I imagine in the future we'll have more faction specific backpacks like a Vault-Tec satchel, BoS backpack, etc.

**Comment by Hekzder** [comment]
I also just pushed a commit that fixes the misspelled military backpack inventory state, fixes a couple missing parent proc calls, and should fix the integration test fails on the backpack thing.

**Comment by TheRealScarHomie** [comment]
ok I would actually consider this finished now

**Review by Hekzder — APPROVED** [review]
Aside from a runtime that we will try to look into when possible, this seems good for a merge but @Koshenko will probably be the one to pull the trigger after he takes a look. Good work Scar.

**Pull request merged**